doc/api.md

```diff
@@ -133,9 +133,9 @@ parameters. The specific parameters that can be configured for a given config gr
 ### Devices
 
 A device contains the information that connects a physical device to a particular entity in the Context Broker. Devices
-are identified by a `device_id`, and they are associated to an existing config group based in `apiKey` matching or
-`type` matching (in the case `apiKey` matching fails). For instance, let's consider a situation in which a config group
-has been provisioned with `type=X`/`apiKey=111` and no other config group has been provisioned.
+are identified by a `device_id`, and they are associated to an existing config group based in `apiKey` matching. For
+instance, let's consider a situation in which a config group has been provisioned with `type=X`/`apiKey=111` and no
+other config group has been provisioned.
 
 The IoT Agents offer a provisioning API where devices can be preregistered, so all the information about service and
```

Comment:
To make this more explicit, maybe we can add a new sub-section after `### Devices` with the following text:

> ### Uniqueness of groups and devices
> 
> * Group service uniqueness is defined by the combination of: service, subservice and apiKey
> * Device uniqueness is defined by the combination of: service, subservice, device_id and apiKey. Note that the several devices with the same device_id are allowed in the same service and subservice as long as their apiKeys are different.

Please, check, just in case I can be wrong with the above text...

(Btw, it's a bit weird that device_id uses snake_case but apiKey uses camelCase. In the database the MongoDB fields use the same?)